src/coredump/coredump.c

```diff
@@ -1087,40 +1096,67 @@ static int process_socket(int fd) {
                         goto finish;
                 }
 
-                /* The final zero-length datagram carries the file descriptor and tells us
+                /* The final zero-length datagram carries the file descriptors and tells us
                  * that we're done. */
                 if (n == 0) {
                         struct cmsghdr *found;
+                        unsigned n_fds = 0;
+
+                        found = cmsg_find(&mh, SOL_SOCKET, SCM_RIGHTS, CMSG_LEN(sizeof(int)));
+                        if (first) {
+                                if (found) {
+                                        /* This is the first message that carries file descriptors, maybe there will be
+                                         * one more that actually contains array of two descriptors. */
+                                        assert(input_fd < 0);
+
+                                        input_fd = *CMSG_TYPED_DATA(found, int);
+                                        first = false;
+
+                                        continue;
+                                } else {
+                                        /* This is zero length message but it has no file descriptor this is the protocol
```

Comment:
this not entirely true, it could also have more than one fd. it has the wrong number, that's what is key here

Reply:
I will adjust the comment.